Sources/MapboxCoreNavigation/LegacyRouteController.swift

```diff
@@ -286,7 +286,7 @@ open class LegacyRouteController: NSObject, Router, InternalRouter, CLLocationMa
     private func checkForUpdates() {
         #if TARGET_IPHONE_SIMULATOR
```

Comment:
This is a C/Objective-C macro. The Swift equivalent is `#if targetEnvironment(simulator)`. None of the code below compiles, hence the reference to VersionProvider below that’s now orphaned. I’m not sure if this check even worked back in #1007 when we introduced it.

Reply:
Once we fix this issue, we can update the changelog to mention that there’s a console message when a new version of the navigation SDK is available.